provisioning/service/src/Manager/IndividualEnrollmentManager.cs

```diff
@@ -184,26 +186,37 @@ internal static class IndividualEnrollmentManager
         ///
         /// <param name="querySpecification">is a <code>string</code> with the SQL query specification. It cannot be <code>null</code>.</param>
         /// <param name="pageSize">the <code>int</code> with the maximum number of items per iteration. It can be 0 for default, but not negative.</param>
-        /// <returns>A <see cref="Query"/> iterator.</returns>
+        /// <returns>A <see cref="QueryManager"/> iterator.</returns>
         /// <exception cref="ArgumentException">if the provided parameter is not correct.</exception>
-        internal static Query CreateQuery(QuerySpecification querySpecification, int pageSize = 0)
+        internal static QueryManager CreateQuery(
+            ServiceConnectionString provisioningConnectionString, 
+            QuerySpecification querySpecification, 
+            CancellationToken cancellationToken, 
+            int pageSize = 0)
         {
-            //TODO: Implement.
-
             /* SRS_INDIVIDUAL_ENROLLMENT_MANAGER_21_014: [The CreateQuery shall throw ArgumentException if the provided querySpecification is null.] */
-            /* SRS_INDIVIDUAL_ENROLLMENT_MANAGER_21_015: [The CreateQuery shall return a new Query for IndividualEnrollments.] */
+            if(querySpecification == null)
+            {
+                throw new ArgumentNullException("querySpecification cannot be null");
+            }
 
-            throw new NotSupportedException("Query is not supported yet");
+            if(pageSize < 0)
+            {
+                throw new ArgumentException("pageSize cannot be negative");
```

Comment:
Please use `nameof()` everywhere the name of a parameter is required. Allows keeping the code and text in-sync. End sentences in ".".
`$"{nameof(pageSize)} cannot be negative."`

Reply:
done